### ISSUE TITLE
Xcode 6.3.1 Linker Error

### ISSUE THREAD

**Issue by bdkjones** [Kentzo/ShortcutRecorder#39]
I updated Xcode from version 6.1 to 6.3.1 last night and now I get the following Linker error when I try to build a project that uses ShortcutRecorder:

![screen shot 2015-05-03 at 13 46 07](https://cloud.githubusercontent.com/assets/926566/7446850/e7725bb4-f19a-11e4-9187-7f0e40a1e70a.png)

I have updated to the latest release of ShortcutRecorder and have followed your instructions exactly to include it in my project. This was working perfectly for months before I updated Xcode, but now I can't for the life of me figure out how to resolve this Linker error.

Does anyone have an idea?

**Comment by bdkjones** [comment]
Well, I fixed the problem, but I'm unsure whether my fix is a good, permanent one. 

I tracked down the duplicate symbol to `SRKeyCodeTransformer.h`:

```
NS_ENUM(unichar, SRKeyCodeGlyph)
{
...
}
```

I just moved this enumeration to `SRKeyCodeTransfomer.m` instead of the header (since it didn't appear to be used anywhere other than this implementation file). This resolved the Linker error and ShortcutRecorder appears to still work just fine.

I have no idea why this turned into a duplicate symbol in the first place.

**Comment by Kentzo** [comment]
@bdkjones Interesting. Have you tried clean build?

**Comment by bdkjones** [comment]
Roughly 6,578 times.

**Comment by Kentzo** [comment]
@bdkjones Please try to build from develop. Looks like you catched a bug :)

**Comment by bdkjones** [comment]
Son of a....

You know, I spent HOURS today trying to figure out why the Linker was calling that a duplicate symbol. It just didn't make sense; I use NS_ENUM all the time in headers that get imported into countless files. So I finally gave up and just moved the thing to the `*.m` file out of frustration, which is why I said I didn't think that was a very good fix. 

All for a missing `typedef`. 

You might consider filing an enhancement request with Apple. When someone uses NS_ENUM without `typedef`, Clang should probably flag that because it's almost guaranteed to be a mistake.

Thanks!
